### ISSUE TITLE
Need a way to leave full-screen mode

### ISSUE THREAD

**Issue by philgoetz** [AppleWin/AppleWin#1188]
There's an icon you can click, and I think a function key also, that will change AppleWin to run in full-screen mode.
There is, as far as I can tell, sometimes no way to get it back into windowed mode.
All of the icons on the right edge of the screen are mostly off of the right edge of the window in full-screen mode.
That itself is a bug worth fixing, or would be if I ever wanted to use full-screen mode, which I never have.
But sometimes, clicking on the icon that changes screen size will restore it to windowed mode; sometimes, it is simply stuck in fullscreen mode & there's nothing to do but to kill the program and restart it.
It would be great to have a way to switch between fullscreen and windowed mode under the F8 options menu.

**Comment by tomcw** [comment]
F6 corresponds to the "Windowed / Full Screen" icon (on the right hand side).

ALT+ENTER _should_ also toggle full-screen, but there's currently a bug (regressed at 1.30.0) where it defaults to allowing the emulated program to be able to read `Open Apple` and `Enter` keys.

(See the help file, under "Command Line": `-alt-enter=<toggle-full-screen|open-apple-enter>`)

**Comment by tomcw** [comment]
>All of the icons on the right edge of the screen are mostly off of the right edge of the window in full-screen mode.

I can only reproduce this when I have VidHD enabled (in slot 3).
With VidHD (in full screen mode), then I can't see any of the Toolbar icons, nor can I see any drive/keyboard status.

**Comment by tomcw** [comment]
> ...there's nothing to do but to kill the program and restart it.

Yes, I can imagine a poor user that's enabled full screen (eg. clicked the "Windowed / Full Screen" icon), and then can't find a way to revert back to windowed mode.

If I fix the VidHD/Full-screen issue (above), then the user (with enough movement of the mouse cursor) should hopefully discover the Toolbar icons on the right, and be able to click the icon to revert to windowed mode.

Or maybe better: if you move the mouse cursor to the top of the screen, then it reveals an "Exit Full Screen Mode" button.

**Comment by philgoetz** [comment]
> > All of the icons on the right edge of the screen are mostly off of the right edge of the window in full-screen mode.
> 
> I can only reproduce this when I have VidHD enabled (in slot 3). With VidHD (in full screen mode), then I can't see any of the Toolbar icons, nor can I see any drive/keyboard status.

I should have mentioned that my monitor is mounted vertically, with resolution 1440 x 2560.

**Comment by tomcw** [comment]
NB. [AppleWin 1.30.14.0](https://github.com/AppleWin/AppleWin/releases/tag/v1.30.14.0) release fixes the regression with ALT+ENTER (by default, it now toggles full screen).

**Comment by philgoetz** [comment]
On Sun, Mar 5, 2023 at 5:28 PM TomCh ***@***.***> wrote:

> NB. AppleWin 1.30.14.0
> <https://github.com/AppleWin/AppleWin/releases/tag/v1.30.14.0> release
> fixes the regression with ALT+ENTER (by default, it now toggles full
> screen).Message ID: ***@***.***>
>
It would be nice if alt+enter cycled through all the screen resolutions,
from 1x (which is 2x actual Apple screen dimensions, I think?), to 2x, to
full-screen.

Since you've done the work to scale the display to full-screen, though, it
seems you could adapt this to give more sizes, like 1.5x.

**Comment by tomcw** [comment]
FYI:
- F6 toggles between full screen & windowed mode (so same as ALT+ENTER).
- CTRL+F6 cycles through the windowed mode scale factors: currently just 1x and 2x.

>1x (which is 2x actual Apple screen dimensions, I think?)

1x is 560x384... where 560 is the real Apple II width in DHIRES/double-hires pixels, and 384 is twice the Apple II's vertical resolution - to maintain aspect ratio.

**Comment by philgoetz** [comment]
On Mon, Mar 6, 2023 at 3:05 PM TomCh ***@***.***> wrote:
>
> FYI:
>
> F6 toggles between full screen & windowed mode (so same as ALT+ENTER).
> CTRL+F6 cycles through the windowed mode scale factors: currently just 1x and 2x.
>
> 1x (which is 2x actual Apple screen dimensions, I think?)
>
> 1x is 560x384... where 560 is the real Apple II width in DHIRES/double-hires pixels, and 384 is twice the Apple II's vertical resolution - to maintain aspect ratio.

Oh, right.  I tend to think "280", because that's how it was
advertised, and how e.g. Applesoft thinks of it; but it's really a
hybrid of 280 and 560 pixel resolution.

Have you ever thought about making a color mode in which pixels have
color (other than white) only if there's an empty pixel on each side
of them?  That would eliminate the color borders around text displayed
using hi-res mode.  I realize this may not be worth doing; it doesn't
have very many use cases.

**Comment by tomcw** [comment]
The HIRES "280" pixel mode is just a special case of the DHIRES "560" pixel mode.
IE. the HIRES mode does double up on the DHIRES pixels (+ the HIRES byte's high bit shifts the pixels by 1 "560" pixel to shift the colour phase, and get 2 more HIRES colours).

>Have you ever thought about making a color mode in which pixels have
>color (other than white) only if there's an empty pixel on each side of them?

In a very old AppleWin (v1.25 being the last one) there was a "Color (text optimized)" mode:
"This mode does a bit of image processing to make text more readable on the hires screen."

IMO, it was too niche (and I wanted to cut down on the number of display modes). If people want sharp TEXT (or sharp text drawn in HIRES), then the real RGB (or Composite?) Monitor modes are a better fit. And it's easy enough to use F9 (or CTRL+F9) to cycle to a better display mode, depending on your current case.

**Comment by Michaelangel007** [comment]
I still miss the Text Legacy mode myself. :-/

The "proper" solution would entail three things:

1. Leave all the video modes in.
2. Have a user-defined list of which modes they want to cycle between. i.e I NEVER use the TV modes and would like to skip them when pressing F6:

    All Video Modes      F6 Cycle Modes
    Color (Composite)    Color (NTSC)
    Color (RGB Card)     Color (Text)
    Color TV             Monochrome (Green)
    B&W TV               
    Monochrome (Amber)   
    Monochrome (White)   
    [-->]                [<--]

3. Remove the two monochrome modes from the default F6 Cycle list.
